authorize.go

```diff
@@ -6,16 +6,109 @@
 
 package main
 
-import "net/http"
+import (
+	"crypto/sha256"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/mozilla-services/hawk-go"
+)
+
+const maxauthage time.Duration = 60 * time.Second
 
 // authorize validates the hawk authorization header on a request
 // and returns the userid and a boolean indicating authorization status
-func (a *autographer) authorize(r *http.Request) (userid string, authorize bool, err error) {
-	return "", true, nil
+func (a *autographer) authorize(r *http.Request, body []byte) (userid string, authorize bool, err error) {
+	var (
+		auth *hawk.Auth
+	)
+	if r.Header.Get("Authorization") == "" {
+		return "", false, fmt.Errorf("missing Authorization header")
+	}
+	auth, err = hawk.ParseRequestHeader(r.Header.Get("Authorization"))
+	if err != nil {
+		return
+	}
+	if time.Now().Sub(auth.Timestamp) > maxauthage {
```

Comment:
It's not mandatory because Go knows how to do the timezone translation, but I agree it is cleaner. Hawk specifies the timestamp as `the number of seconds since January 1, 1970 00:00:00 GMT`, and the `hawk-go` library uses Go's `Unix()` time to generate it, so it's always UTC.